vms/avm/txs/txstest/backend.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package txstest
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/chains/atomic"
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/vms/avm/state"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/wallet/chain/x/builder"
+	"github.com/ava-labs/avalanchego/wallet/chain/x/signer"
+)
+
+const maxPageSize uint64 = 1024
+
+var (
+	_ builder.Backend = (*walletBackendAdapter)(nil)
+	_ signer.Backend  = (*walletBackendAdapter)(nil)
+)
+
+func NewBackend(
+	ctx *snow.Context,
+	state state.State,
+	sharedMemory atomic.SharedMemory,
+	codec codec.Manager,
+) *Backend {
+	return &Backend{
+		xchainID:     ctx.XChainID,
+		state:        state,
+		sharedMemory: sharedMemory,
+		codec:        codec,
+	}
+}
+
+type Backend struct {
```

Comment:
Can we call this `UTXOs` or something similar to make it more obvious that this is used for the wallet's utxo state

Reply:
Done, but made it unexported